config/dev-modules.yaml

```diff
@@ -3,6 +3,9 @@
 ##
 modules:
   mod_blocking: {} # requires mod_privacy
+  mod_block_incoming: {
+    blocking_users: ["test2@localhost", "test3@localhost", "11339339@chat.skillz.com", "20361620@chat.skillz.com", "12669429@chat.skillz.com", "97789891@chat.skillz.com", "67115158@chat.skillz.com", "40890859@chat.skillz.com", "50447625@chat.skillz.com", "26581865@chat.skillz.com", "41241278@chat.skillz.com", "14462266@chat.skillz.com", "33411470@chat.skillz.com", "29450014@chat.skillz.com", "32704697@chat.skillz.com", "15643249@chat.skillz.com", "15550929@chat.skillz.com", "20842529@chat.skillz.com", "48178115@chat.skillz.com", "62393530@chat.skillz.com", "37172470@chat.skillz.com", "19925798@chat.skillz.com", "68727473@chat.skillz.com", "18303590@chat.skillz.com", "47350879@chat.skillz.com", "71665163@chat.skillz.com", "25325510@chat.skillz.com", "47907443@chat.skillz.com", "21710348@chat.skillz.com", "25339760@chat.skillz.com", "12959623@chat.skillz.com", "41438543@chat.skillz.com", "41481919@chat.skillz.com", "72621740@chat.skillz.com", "16242034@chat.skillz.com", "61225504@chat.skillz.com", "18456864@chat.skillz.com", "14688690@chat.skillz.com", "8817662@chat.skillz.com", "38566579@chat.skillz.com", "11338313@chat.skillz.com", "38596512@chat.skillz.com", "46910420@chat.skillz.com", "47060194@chat.skillz.com", "59207162@chat.skillz.com", "18051462@chat.skillz.com", "25000144@chat.skillz.com", "22405075@chat.skillz.com", "18099391@chat.skillz.com", "66611346@chat.skillz.com", "66361397@chat.skillz.com", "12877394@chat.skillz.com"]
```

Comment:
I'm thinking we don't need to store the jid and can just be the "nick" and append the host in code since it will always be the same env host

Would there be any limitations to storing super long lists in this format? I know we don't expect this list to grow massively and this is a cool way to easily refresh the list

Reply:
A followup thought, do the HTTP admin commands have any method of persisting here? Or would we be doing a 2-step of persisting it here, then running the command the update it live?

Reply:
To be a generic module I think we need to keep the host since ejabberd can have multiple in general, but for Skillz purposes yes that makes sense. There could be limitations if we are iterating the whole list (or half on avg) to determine membership. I will switch to a dict for better performance. This list would remain the same but the gen server would leverage a dict for its state. The http commands will update the state (list or dict) in memory but it won't persist here. I am envisioning that we would want to have a way to quickly add / remove a mod using these commands and then later at off hours we'd generate the list and set it in this config with a publish so it's persisted.